lib/spack/spack/stage.py

```diff
@@ -775,8 +778,11 @@ def get_checksums_for_versions(
                  for v in sorted_versions]))
     print()
 
-    archives_to_fetch = tty.get_number(
-        "How many would you like to checksum?", default=1, abort='q')
+    if not batch:
+        archives_to_fetch = tty.get_number(
+            "How many would you like to checksum?", default=1, abort='q')
+    else:
+        archives_to_fetch = len(sorted_versions)
```

Comment:
I wonder if this would not be better factored by moving the user interaction out to a separate function in `lib/spack/cmd`. It's also true that this can be done in a separate PR. @tgamblin Do you have opinions on this?

Reply:
Do we use this specific interaction anywhere else? If not - this is not worth moving out IMHO. I think the only change would be to replace inverted condition with regular.